### PR TITLE
Add role for kubectl top pod

### PR DESCRIPTION
**Pull request by CalleB3**
Added role to enable our developers to use "kubectl top pod" command.

### CODE REVIEW

**Comment by CalleB3** [comment]
Output from "make plan", censored some info and replace with "XXXXX".
@NissesSenap 
Terraform will perform the following actions:

  # module.aks1_core.kubernetes_cluster_role.top will be created
  + resource "kubernetes_cluster_role" "top" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + labels           = {
              + "xkf.xenit.io/kind" = "platform"
            }
          + name             = "top"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + rule {
          + api_groups = [
              + "rbac.authorization.k8s.io",
            ]
          + resources  = [
              + "pods",
            ]
          + verbs      = [
              + "get",
              + "list",
              + "watch",
            ]
        }
    }

  # module.aks1_core.kubernetes_role_binding.top["apps-svc"] will be created
  + resource "kubernetes_role_binding" "top" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + labels           = {
              + "aad-group-name"    = "SG-AAD-AKS-XXXXXXXXXXXXXXXXX"
              + "xkf.xenit.io/kind" = "platform"
            }
          + name             = "apps-svc-top"
          + namespace        = "apps-svc"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + role_ref {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "ClusterRole"
          + name      = "top"
        }

      + subject {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "Group"
          + name      = "c551cf11-XXXXXXXXXXXXX"
          + namespace = "default"
        }
    }

  # module.aks1_core.kubernetes_role_binding.top["lab"] will be created
  + resource "kubernetes_role_binding" "top" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + labels           = {
              + "aad-group-name"    = "SG-AAD-AKS-XXXXXXXXXXXXXXXXXX"
              + "xkf.xenit.io/kind" = "platform"
            }
          + name             = "lab-top"
          + namespace        = "lab"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + role_ref {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "ClusterRole"
          + name      = "top"
        }

      + subject {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "Group"
          + name      = "9881edc1-XXXXXXXXXXXXXXXXX"
          + namespace = "default"
        }
    }

  # module.aks1_core.kubernetes_role_binding.top["monitor"] will be created
  + resource "kubernetes_role_binding" "top" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + labels           = {
              + "aad-group-name"    = "SG-AAD-AKS-XXXXXXXXXX
              + "xkf.xenit.io/kind" = "platform"
            }
          + name             = "monitor-top"
          + namespace        = "monitor"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + role_ref {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "ClusterRole"
          + name      = "top"
        }

      + subject {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "Group"
          + name      = "dcecde4f-XXXXXXXXXXXXX"
          + namespace = "default"
        }
    }

  # module.aks1_core.kubernetes_role_binding.top["xks"] will be created
  + resource "kubernetes_role_binding" "top" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + labels           = {
              + "aad-group-name"    = "SG-AAD-AKS-XXXXXXXX"
              + "xkf.xenit.io/kind" = "platform"
            }
          + name             = "xks-top"
          + namespace        = "xks"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + role_ref {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "ClusterRole"
          + name      = "top"
        }

      + subject {
          + api_group = "rbac.authorization.k8s.io"
          + kind      = "Group"
          + name      = "15c217c1-3XXXXXXXXXXXXX"
          + namespace = "default"
        }
    }

File: modules/kubernetes/aks-core/k8s-cluster-role.tf [hunk 1]
```diff
@@ -46,3 +46,17 @@ resource "kubernetes_cluster_role" "custom_resource_edit" {
...
+
+resource "kubernetes_cluster_role" "top" {
+  metadata {
+    name = "top"
+    labels = {
+      "xkf.xenit.io/kind" = "platform"
+    }
+  }
+  rule {
+    api_groups = ["rbac.authorization.k8s.io"]
```
Comment (NissesSenap) [inline modules/kubernetes/aks-core/k8s-cluster-role.tf:58]: I think it should say: `metrics.k8s.io`

File: modules/kubernetes/eks-core/k8s-role-binding.tf [hunk 2]
```diff
@@ -139,3 +139,24 @@ resource "kubernetes_role_binding" "sa_helm_release" {
...
+    namespace = kubernetes_namespace.tenant[each.key].metadata[0].name
+    labels = {
+      "aad-group-name"    = var.aad_groups.edit[each.key].name
+      "xkf.xenit.io/kind" = "platform"
+    }
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "top"
```
Comment (NissesSenap) [inline modules/kubernetes/eks-core/k8s-role-binding.tf:155]: Use this way of getting the name instead:
`kubernetes_cluster_role.top.metadata[0].name`

**Review by NissesSenap — CHANGES_REQUESTED** [review]
There are a few changes that I would like to see.

File: modules/kubernetes/eks-core/k8s-role-binding.tf (hunk 2 above)
Comment (CalleB3) [reply modules/kubernetes/eks-core/k8s-role-binding.tf]: Fixed

File: modules/kubernetes/aks-core/k8s-cluster-role.tf (hunk 1 above)
Comment (CalleB3) [reply modules/kubernetes/aks-core/k8s-cluster-role.tf]: Fixed

**Review by NissesSenap — APPROVED** [review]
LGTM

**Pull request merged**